geo/src/algorithm/rotate.rs

```diff
@@ -174,17 +177,24 @@ where
     fn rotate(&self, angle: T) -> Self {
         // if a polygon has holes, use the centroid of its outer shell as the rotation origin
```

Comment:
I realize this predates your work, but I'm surprised by the behavior documented in this comment. I would have guessed we're always rotating about the centroid (including the holes). @urschrei - do you recall why we do it this way?

Reply:
My commit comment (https://github.com/georust/geo/commit/dd345c5491a09b3e45f3c08c0bc3a78f7d627bb9) isn't super informative, but I assume it's because holes shift the centroid and that's not what you want when rotating. I'll have to confirm my suspicion that Shapely / GEOS do this, and I copied them, however.

Reply:
> it's because holes shift the centroid and that's not what you want when rotating.

I brought this up because I think rotating around the geometry's centroid (accounting for holes - the "real" centroid) in all cases is what I'd want and expect. 

I think primarily because this is a concise and consistent explanation of what the method does.

I think I see what you're saying about how some people might want to rotate around something more like a bbox center, rather than a centroid, but this "exterior-only centroid" concept seems a little uncommon.


From shapely, their rotate method takes a "point of origin" argument with docs:

>  The point of origin can be a keyword 'center' for the bounding box center (default), 'centroid' for the geometry’s centroid, a Point object or a coordinate tuple (x0, y0).

I'd love to steal this design, though because we don't have default args the api might be a little different... something like:

```
trait Rotate {
  #[deprecate(note="use more specific rotate_around_centroid or rotate_around_center")]
  fn rotate(&self, angle: T) -> Self {
    rotate_around_centroid(angle)
  }

  fn rotate_around_coord(&self, coord: Coordinate<T>, angle: T) -> Self;

  /// equivalent to:
  /// ```
  /// let center = g.bounding_rect().center();
  /// g.rotate_around_coord(center, angle)
  /// ```
  fn rotate_around_center(&self, angle: T) -> Self;

  /// ```
  /// let centroid = g.centroid();
  /// g.rotate_around_coord(centroid, angle)
  /// ```
  fn rotate_around_centroid(&self, angle: T) -> Self;
}
```

What would you think of something like that?

Reply:
I am def. in favour. I just had a look to see whether we could do something tricky with an enum and default value (center), but it seems a bit too messy atm – separate methods is a cleaner design.

Reply:
Ok, I'm gonna make a new issue with these details, linking to this discussion, and give @mapoulos their PR back. 😅